### PR TITLE
NIAD-1117: Make identifier and healthcare identifier code fields in RequesterNameAndAddress optional

### PR DESCRIPTION
**Pull request by n-r-b**
## Description

Made the `identifier` and `healthcareRegistrationIdentificationCode` fields optional in RequesterNameAndAddress. 

Example of NAD+PO with an identifier & healthcare code: 
`NAD+PO+G8040642:900++HOLMES`

Example of NAD+PO without identifier & healthcare code: 
`NAD+PO+++HOLMES`

## Jira Ticket

https://gpitbjss.atlassian.net/browse/NIAD-1117

## Checklist

These are items (excluding GitHub Checks) which should be confirmed before a branch is ready to merge.

- [x] Acceptance Criteria met
- [x] Commit messages are meaningful
- [x] Manually tested
- [x] Self-reviewed your code
- [ ] Code reviewed from two other developers
- [x] If your pull request depends on any other, please link them in the description
- [x] main branch is passing/green on CI
- [x] Add/update any relevant documentation

### CODE REVIEW

File: src/intTest/resources/fhir/multi_pathology_msg3.fhir.json [hunk 1]
```diff
@@ -43,13 +43,12 @@
         "id": "1896e090-962b-4c66-aa41-8154e4623b49",
         "identifier": [
           {
-            "system": "https://fhir.nhs.uk/Id/sds-user-id",
-            "value": "G8040642"
+            "system": "https://fhir.nhs.uk/Id/sds-user-id"
```
Comment (n-r-b) [inline src/intTest/resources/fhir/multi_pathology_msg3.fhir.json:46]: This shouldn't be here when there's no identifier

File: src/test/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/PractitionerMapperTest.java [hunk 2]
```diff
@@ -68,6 +67,35 @@ void testMapMessageToRequestingPractitionerWithRequester() {
...
+
+        Optional<Practitioner> requestingPractitioner = mapper.mapToRequestingPractitioner(message);
+        assertThat(requestingPractitioner).isNotEmpty()
+            .hasValueSatisfying(practitioner -> assertAll(
+                () -> assertThat(practitioner.getName())
+                    .hasSize(1)
+                    .first()
+                    .extracting(HumanName::getText)
+                    .isEqualTo("Alan Turing"),
+                () -> assertThat(practitioner.getIdentifier())
```
Comment (n-r-b) [inline src/test/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/PractitionerMapperTest.java:89]: This shouldn't be here

File: src/intTest/resources/fhir/multi_pathology_msg3.fhir.json (hunk 1 above)
Comment (n-r-b) [reply src/intTest/resources/fhir/multi_pathology_msg3.fhir.json]: Updated logic in PractitionerMapper to only map identifier when it exists

File: src/test/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/PractitionerMapperTest.java (hunk 2 above)
Comment (n-r-b) [reply src/test/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/PractitionerMapperTest.java:89]: Removed

File: src/test/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/PractitionerMapperTest.java [hunk 3]
```diff
@@ -68,6 +67,30 @@ void testMapMessageToRequestingPractitionerWithRequester() {
...
+                    .hasSize(0)
```
Comment (andrewwa-kainos) [inline src/test/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/PractitionerMapperTest.java:90]: This is just
```suggestion
                    .isEmpty()
```

**Review by andrewwa-kainos — APPROVED** [review]

**Review by matthewdoreilly — DISMISSED** [review]

**Review by matthewdoreilly — APPROVED** [review]

**Pull request merged**